brigade-worker/src/k8s.ts

```diff
@@ -487,20 +509,35 @@ export class JobRunner implements jobs.JobRunner {
    * update pod info on event using watch
    */
   private startUpdatingPod(): any {
-    return watchClient.watch(
-      `/api/v1/namespaces/${this.project.kubernetes.namespace}/pods`,
-      { labelSelector: `build=${this.event.buildID},jobname=${this.job.name}` },
-      (_type, obj) => {
-        this.pod = obj;
-      },
-      err => {
-        if (err) {
-          this.logger.error("failed pod lookup");
-          this.logger.error(err);
-          this.cancel = true;
+    const url = `${kc.getCurrentCluster().server}/api/v1/namespaces/${this.project.kubernetes.namespace}/pods`;
+    const requestOptions = {
+      qs: { watch: true, labelSelector: `build=${this.event.buildID},jobname=${this.job.name}` },
+      method: 'GET', uri: url, useQuerystring: true, json: true,
+    };
+    kc.applyToRequest(requestOptions);
+    const stream = new byline_1.LineStream();
+    stream.on('data', (data) => {
+      let obj = null;
+      try {
+        if (data instanceof Buffer) {
+          obj = JSON.parse(data.toString());
         }
+        else {
+          obj = JSON.parse(data);
+        }
+      } catch (e) { } //let it stay connected.
+      if (obj && obj.object) {
+        this.pod = obj.object;
```

Comment:
Tying in to the comment above about having a typed `pod` here.

Reply:
I also updated it with the proper type, V1Pod. Thanks for pointing out.